.github/workflows/check.yml

```
@@ -32,44 +31,61 @@ jobs:
         - 3.8
         - 3.7
         - 3.6
+        - 3.5
+        - pypy3
         - 2.7
         - pypy2
-        - pypy3
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: tox-dev/action@v1.0.0
-      with:
-        python-version: ${{ matrix.py }}
-
+    - name: install OS dependencies
+      run:   |
+        if [ "$RUNNER_OS" == "Linux" ]; then
+             sudo apt-get update -y && sudo apt-get install fish csh
+        elif [ "$RUNNER_OS" == "Windows" ]; then
+             choco install vcpython27 --yes -f
+        elif [ "$RUNNER_OS" == "macOS" ]; then
+             brew update -vvv && brew install fish tcsh
+        fi
+      shell: bash
```

I personally prefer these as separate steps so that they are exposed better in the log:
```suggestion
    - name: install GNU/Linux dependencies
      if: runner.os == 'Linux'
      run: sudo apt-get update -y && sudo apt-get install fish csh
    - name: install macOS dependencies
      run: brew update -vvv && brew install fish tcsh
      if: runner.os == 'macOS'
    - name: install Windows dependencies
      run: choco install vcpython27 --yes -f
      if: runner.os == 'Windows'
```

---

Yeah had that first but this makes it clearer that only one of them is active at a time 🤔😎

---

if-clause makes other steps be marked grey so usually it's okay combined with the os in the step name

---

I prefer no step present over gray step 😉 also decreases the step count this way